src/analysis/retail/druid/restoration/modules/spells/NaturesVigil.tsx

```diff
@@ -0,0 +1,154 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { Options } from 'parser/core/Module';
+import { TALENTS_DRUID } from 'common/TALENTS';
+import Events, { CastEvent, DamageEvent, HealEvent } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { formatNumber, formatPercentage } from 'common/format';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import { SINGLE_TARGET_HEALING } from 'analysis/retail/druid/restoration/constants';
+import { SpellLink } from 'interface';
+
+// separate class for Resto because it converts healing to damage rather than other way around
+/**
+ * **Nature's Vigil**
+ * Class Talent
+ *
+ * All single target healing also damages a nearby enemy target for 20% of the healing done.
+ */
+class NaturesVigil extends Analyzer {
+  /** Trackers for info about each cast */
+  nvTracker: NaturesVigilCast[] = [];
+  /** Mapping by number of the contributions to NV's damage */
+  stHealingDuringNv: { [spellId: number]: { id: number; amount: number } } = {};
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_DRUID.NATURES_VIGIL_TALENT);
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_DRUID.NATURES_VIGIL_TALENT),
+      this.onNvCast,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.NATURES_VIGIL_DAMAGE),
+      this.onNvDamage,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SINGLE_TARGET_HEALING),
+      this.onStHeal,
+    );
+  }
+
+  onNvCast(event: CastEvent) {
+    this.nvTracker.push({
+      timestamp: event.timestamp,
+      damage: 0,
+    });
+  }
+
+  onNvDamage(event: DamageEvent) {
+    const currTracker = this.nvTracker.at(-1);
+    if (currTracker) {
+      currTracker.damage += event.amount + (event.absorbed || 0);
+    }
+  }
+
+  onStHeal(event: HealEvent) {
+    if (this.selectedCombatant.hasBuff(TALENTS_DRUID.NATURES_VIGIL_TALENT.id)) {
+      const spellId = event.ability.guid;
+      // if (spellId === SPELLS.REJUVENATION_GERMINATION.id) {
+      //   spellId = SPELLS.REJUVENATION.id;
+      // } else if (spellId === SPELLS.LIFEBLOOM_UNDERGROWTH_HOT_HEAL.id || spellId === SPELLS.LIFEBLOOM_BLOOM_HEAL.id) {
+      //   spellId = SPELLS.LIFEBLOOM_HOT_HEAL.id;
+      // }
```

Comment:
stray comments

Reply:
oh woops - yeah I was going to coalesce similar spells, then decided it wasn't worth the trouble